set.mm

```diff
@@ -467069,6 +467273,448 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
+      EUIUJZUKUJZULZUMZLFIUNUJZUOJIHUDZUEWLJGBIBUFZWKUDZWSEUJUPZUKUJZULZUMZLWQU
+      OACDELWHIFGHJKMNOPQRAIICWHIIWHUQZIIWHURZAIUSZIIWHUTZVATVBSAICVCCVCUEACGVD
+      NVEVFTVGZVHAWJWRJAWIIHIIVKWIIULAIVLIIVIVAVJVMAWPXDLWQAWKWQUEZUMZWOXCWLXKW
+      NXBJXKWMXAGUKXKBIIWTWSEIWKWHVCVCXJWKIVNAWKFIVOVPXKIIWHXEXFXKXGXHVAVQAIVCU
+      EXJXIVRZXLIVSXKWSIUEZUMWTWCXMWSWHUDWSULXKIWSVTVPWAWBWDWEWFWG $.
+  $}
+
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Vector Spaces
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c dim $.
+
+  $( Extend class notation with the dimension of a vector space. $)
+  cldim $a class dim $.
+
+  $( Define the dimension of a vector space as the cardinality of its bases.
+     Note that by ~ lvecdim , all bases are equinumerous.  (Contributed by
+     Thierry Arnoux, 6-May-2023.) $)
+  df-dim $a |- dim = ( f e. _V |-> U. ( # " ( LBasis ` f ) ) ) $.
+
+  ${
+    $d F t x f $.  $d J t x f $.  $d S t x $.
+    dimval.1 $e |- J = ( LBasis ` F ) $.
+    $( The dimension of a vector space ` F ` is the cardinality of one of its
+       bases.  (Contributed by Thierry Arnoux, 6-May-2023.) $)
+    dimval $p |- ( ( F e. LVec /\ S e. J ) -> ( dim ` F ) = ( # ` S ) ) $=
+      ( vf vx vt wcel wa cfv chash cuni wceq cvv cv clbs hashf syl adantr c0 wf
+      clvec cldim cima csn elex syl6eqr imaeq2d unieqd df-dim cn0 cpnf cun wfun
+      fveq2 ffun fvex eqeltri funimaex mp2b uniex fvmpt cen wbr simplll simpllr
+      wral simpr lvecdim syl3anc hasheni eqtr2d wrex ax-mp fvelima mpan r19.29a
+      adantl ralrimiva wb wne lbsex wfn wss ffn fnimaeq0 mp2an necon3bii sylibr
+      ssv eqsn mpbird unisn a1i 3eqtrd ) BUBHZACHZIZBUCJZKCUDZLZAKJZUEZLZXBWPWS
+      XAMZWQWPBNHXEBUBUFEBKEOZPJZUDZLXANUCXFBMZXHWTXIXGCKXIXGBPJZCXFBPUODUGUHUI
+      EUJWTNUKULUEUMZKUAZKUNZWTNHQNXKKUPZKCCXJNDBPUQURUSUTVAVBRSWRWTXCWRWTXCMZF
+      OZXBMZFWTVGZWRXQFWTWRXPWTHZIZGOZKJZXPMZXQGCXTYACHZIZYCIXBYBXPYEXBYBMZYCYE
+      AYAVCVDZYFYEWPWQYDYGWPWQXSYDVEWPWQXSYDVFXTYDVHAYACBDVIVJAYAVKRSYEYCVHVLXS
+      YCGCVMZWRXMXSYHXLXMQXNVNGXPCKVOVPVRVQVSWPXOXRVTZWQWPWTTWAZYIWPCTWAYJCBDWB
+      WTTCTKNWCZCNWDWTTMCTMVTXLYKQNXKKWEVNCWJNCKWFWGWHWIFWTXBWKRSWLUIWQXDXBMZWP
+      YLWQXBAKUQWMWNVRWO $.
+  $}
+
+  ${
+    $d V b $.
+    $( Closure of the vector space dimension.  (Contributed by Thierry Arnoux,
+       18-May-2023.) $)
+    dimcl $p |- ( V e. LVec -> ( dim ` V ) e. NN0* ) $=
+      ( vb clvec wcel cv clbs cfv wex cldim cxnn0 c0 wne eqid lbsex n0 sylib wa
+      chash dimval hashxnn0 adantl eqeltrd ex exlimdv mpd ) ACDZBEZAFGZDZBHZAIG
+      ZJDZUFUHKLUJUHAUHMNBUHOPUFUIULBUFUIULUFUIQUKUGRGZJUGAUHUHMSUIUMJDUFUGUHTU
+      AUBUCUDUE $.
+  $}
+
+  ${
+    0ellsp.1 $e |- .0. = ( 0g ` W ) $.
+    0ellsp.b $e |- B = ( Base ` W ) $.
+    0ellsp.n $e |- N = ( LSpan ` W ) $.
+    $( The group identity element is in all spans.  (Contributed by Thierry
```

Comment:
Maybe it's more natural to simply write "zero" instead of "the group identity", when you're talking about left modules or, more generally, abelian groups.